sdv/sampling/hierarchical_sampler.py

```diff
@@ -198,16 +195,16 @@ def _sample(self, scale=1.0):
             sampled_data[table] = self._sample_rows(synthesizer, num_rows)
 
         # Sample rest of the graph
-        starting_table = root_parents.pop()  # Start at any root table
-        self._sample_table(table_name=starting_table, sampled_data=sampled_data)
+        for root_table in root_parents:
+            self._sample_children(table_name=root_table, sampled_data=sampled_data)
 
         added_relationships = set()
         for relationship in self.metadata.relationships:
             parent_name = relationship['parent_table_name']
             child_name = relationship['child_table_name']
             # When more than one relationship exists between two tables, only the first one
-            # is used to recreate the child tables, so that's the only one that needs to be
-            # added back.
+            # is used to recreate the child tables, so the rest of the foreign key columns
```

Comment:
I don't understand your addition. The comment is trying to explain why `_add_foreign_key_columns` is only called once per parent/child pair.

Reply:
I got confused because it said "only one that needs to be added back". Nothing is added back. One foreign key is already present and the rest are added for the first time. I'll update again